### PR TITLE
Activity Log: filter bar design

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #15192

Adds filter bar design + few enhancements on the calendar.

### To test

1. Open Activity Log
2. Play with date range and activity type filters :)

Please, test by:

1. Running it on iPad and iPhone
2. Light and dark mode

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift [hunk 1]
```diff
@@ -158,22 +162,41 @@ class ActivityListViewController: UIViewController, TableViewContainer, ImmuTabl
...
+        let separator = UIView()
+        separator.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.addSubview(separator)
+        NSLayoutConstraint.activate([
+            separator.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: 16),
+            separator.leftAnchor.constraint(equalTo: scrollView.leftAnchor),
+            separator.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+            separator.heightAnchor.constraint(equalToConstant: 1)
+        ])
+        WPStyleGuide.applyBorderStyle(separator)
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift]: This method is pretty messy. What do you think about doing something like:

```
        filterStackView.alignment = .center
        filterStackView.spacing = Constants.filterStackViewSpacing
        filterStackView.translatesAutoresizingMaskIntoConstraints = false

        let filterIcon = UIImageView(image: UIImage.gridicon(.filter))
        filterIcon.tintColor = .listIcon
        filterIcon.heightAnchor.constraint(equalToConstant: Constants.filterHeightAnchor).isActive = true

        filterStackView.addArrangedSubview(filterIcon)
        filterStackView.addArrangedSubview(dateFilterChip)
        filterStackView.addArrangedSubview(activityTypeFilterChip)

        let scrollView = UIScrollView()
        scrollView.canCancelContentTouches = true
        scrollView.addSubview(filterStackView)
        containerStackView.addArrangedSubview(scrollView)

        NSLayoutConstraint.activate([
            filterStackView.leftAnchor.constraint(equalTo: scrollView.leftAnchor, constant: Constants.filterBarHorizontalPadding),
            filterStackView.rightAnchor.constraint(equalTo: scrollView.rightAnchor, constant: Constants.filterBarHorizontalPadding),
            filterStackView.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: Constants.filterBarVerticalPadding),
            filterStackView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: Constants.filterBarVerticalPadding),
            scrollView.heightAnchor.constraint(equalTo: filterStackView.heightAnchor, constant: 2 * Constants.filterBarVerticalPadding)
        ])

        setupDateFilter()
        setupActivityTypeFilter()

        let separator = UIView()
        separator.translatesAutoresizingMaskIntoConstraints = false
        scrollView.addSubview(separator)
        NSLayoutConstraint.activate([
            separator.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: 16),
            separator.leftAnchor.constraint(equalTo: scrollView.leftAnchor),
            separator.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
            separator.heightAnchor.constraint(equalToConstant: 1)
        ])
```

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Heya, looking good. I love the speed of the calendar now 👍 

<details>
 <summary>Do we need to show the scrollbars here?</summary>

</details>

<details>
 <summary>RTL: I think the filter icon should be fully aligned to the right of the view</summary>

</details>

<details>
 <summary>RTL: There are some overlapping issues</summary>

</details>

**Comment by osullivanchris** [comment]
Very slick, great job. A couple of really nice touches. The smart reselection of dates is great (adapting the start/end date whichever I'm closer to). Loading state works well. Generally feels very polished.

Minor details to consider:
- its per design, but does the 'x' in the chip feel too heavy? Should we go with the grey we're using in icons?
- Should we have a shadow beneath the selected dates on scroll? I never love when things just disappear behind nothing in the ui :stuck_out_tongue:
- I noticed on landscape that when I selected one date, the layout for the date selection heading was off-center
- Also noticed scrollbar as @emilylaguna  mentioned

I haven't gotten the build working on my iPad yet but thought I'd share these. If I notice anything on iPad I'll add it later.

**Comment by jkmassel** [comment]
👋 Howdy! Today's the day we're cutting the `16.4` release.

Because of that, this PR will be bumped to `16.5`. If you need it to be part of `16.4`, please merge it into the `release/16.4` branch and DM me – I'll be happy to cut a new beta release!

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift]: Done in f94c667c2885278562c7ab15209a0abd435f2f35

I also extracted the filter to its own file.

**Comment by leandroalonso** [comment]
@emilylaguna @osullivanchris ready for another review!

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Hey RTL is looking much better! I noticed a couple more things:

In LTR the bottom separator doesn't go all to the way to the end when scrolling:

Screen Shot 2020-12-16 at 10 24 26 AM

In RTL there is no separator at all 🤔 
Screen Shot 2020-12-16 at 10 26 12 AM

**Comment by leandroalonso** [comment]
@emilylaguna fixed in cedc3b153b4a559dc5d4f62c862b955279b42111

**Review by emilylaguna — APPROVED** [review]
👏 Nice, working very well now.

**Comment by osullivanchris** [comment]
> @emilylaguna @osullivanchris ready for another review!

All looks great now

- 'x' on the chip works better slightly lighter
- really nice job on the fade-out behind the calendar. Looks slick
- landscape centring issue fixed
- scrollbar hidden, looks much better

Noting you have a kinda animation going I think when the chip changes. I wasn't expecting it, but I think I like it. 

Have not been able to view directly on iPad, but looks good based on screenshots.

Big 👍  from me.

**Pull request merged**